### PR TITLE
Optimize GDScript VM codegen for MSVC

### PR DESCRIPTION
**Pull request by RandomShaper**
This allegedly leads to less optimal code than in the case of GCC/Clang's computed gotos, but is an improvement for those building with MSVC. I haven't benchmarked myself, so TIWAGOS.

### CODE REVIEW

**Comment by RandomShaper** [comment]
> That reminds me, could we use `likely()`/`unlikely()` hints with GCC/Clang here? We do this in error macros already.

If you mean using them for dispatching opcodes, there's the computed gotos optimization already in place for them, for which there's no point in mixing with (`un`)`likely`. Or are you talking about other spot?

**Comment by akien-mga** [comment]
Would be interesting to check how this affects build time for this file, both with and without exceptions enabled (see #80513).

File: modules/gdscript/gdscript_vm.cpp [hunk 1]
```diff
@@ -397,7 +397,13 @@ void (*type_init_function_table[])(Variant *) = {
 #define OPCODES_END
 #define OPCODES_OUT
 #define DISPATCH_OPCODE continue
+#ifdef MSVC
```
Comment (akien-mga) [inline modules/gdscript/gdscript_vm.cpp:400]: Does this work? I think usually we use `_MSC_VER`.

I see only one occurrence of `#ifdef MSVC` in the codebase, which might be a bug.

Comment (bruvzg) [reply modules/gdscript/gdscript_vm.cpp:400]: It is defined:
https://github.com/godotengine/godot/blob/df0a822323a79e1a645f0c6a17d51c7602f23166/platform/windows/detect.py#L382-L388

Comment (akien-mga) [reply modules/gdscript/gdscript_vm.cpp:400]: It seems a bit unnecessary to define it here, to use it only in a single location, when we can rely on `_MSC_VER` like 500 other locations in the codebase :)

So I'd suggest removing that define.

Comment (RandomShaper) [reply modules/gdscript/gdscript_vm.cpp]: Pushed with use of `_MSC_VER`.

**Review by akien-mga — APPROVED** [review]
Looks good to me.

Would like to have some numbers so we know exactly what we gain, but it's not blocking.

**Comment by akien-mga** [comment]
Thanks!

**Comment by Faless** [comment]
For posterity, I did run some benchmarks on this.

The MSVC build is much faster with this patch but still noticeably slower than the GCC/MinGW version using the synthetic test in #11518 adapted to 4.x ( [test.zip](https://github.com/godotengine/godot/files/12718651/test.zip) ).

Here are the hyperfine results for 549fcce5f8f7beace3e5c90e9bbe4335d4fd1476 (+ patch) built with `target=template_release production=yes`

```bash
$ hyperfine.exe -w 1 -r 5 .\test\msvc_base\test.exe .\test\msvc_opt\test.exe .\test\mingw\test.exe
Benchmark 1: .\test\msvc_base\test.exe
  Time (mean ± σ):     22.410 s ±  0.312 s    [User: 22.251 s, System: 0.040 s]
  Range (min … max):   22.075 s … 22.911 s    5 runs

Benchmark 2: .\test\msvc_opt\test.exe
  Time (mean ± σ):     16.885 s ±  0.078 s    [User: 16.701 s, System: 0.052 s]
  Range (min … max):   16.774 s … 16.960 s    5 runs

Benchmark 3: .\test\mingw\test.exe
  Time (mean ± σ):     15.021 s ±  0.103 s    [User: 14.892 s, System: 0.021 s]
  Range (min … max):   14.915 s … 15.194 s    5 runs

Summary
  .\test\mingw\test.exe ran
    1.12 ± 0.01 times faster than .\test\msvc_opt\test.exe
    1.49 ± 0.02 times faster than .\test\msvc_base\test.exe
```

**Pull request merged**